### PR TITLE
Patch back to MR2+

### PR DESCRIPTION
**Pull request by yjian118**
This PR added a list of commits that want to be patched back to MR2+, not straight cherry-picks, please help me to verify if any commit missed

### CODE REVIEW

File: controllers/platformnetwork_controller.go [hunk 1]
```diff
@@ -317,6 +420,18 @@ func (r *PlatformNetworkReconciler) ReconcileAddressPool(client *gophercloud.Ser
...
 		if pool == nil {
 			pool, err = r.ReconcileNewAddressPool(client, instance)
+		} else if instance.Spec.Type == cloudManager.OAMNetworkType {
+			oamNetworkList, err := oamNetworks.ListNetworks(client)
+			if err != nil {
+				err = perrors.Wrapf(err, "failed to get oam network")
+				return err
+			}
+			oamNetwork := &oamNetworkList[0]
+			err = r.ReconcileUpdatedOAMNetwork(client, instance, oamNetwork)
```
Comment (sriram-gn) [inline controllers/platformnetwork_controller.go:430]: Does it mean we are supporting OAM network reconfiguration here?

**Review by hugonicodemos — APPROVED** [review]

File: controllers/platformnetwork_controller.go (hunk 1 above)
Comment (yjian118) [reply controllers/platformnetwork_controller.go:430]: Yes, that's a part of Hugo's fix. OAM network reconfiguration is supported in the current release, I would prefer to pick it up

**Comment by yjian118** [comment]
Tests done:
1 DC deployment
2 subcloud day-2 operation with base profile
3 subcloud day-2 operation with admin network muti-netting assignment
4 subcloud day-2 operation on locked host

**Review by sriram-gn — APPROVED** [review]
Wondering if this  fix is required too:
https://github.com/Wind-River/cloud-platform-deployment-manager/commit/e369ef5a66318156ece65c74c58ee54f046e9f0e

Rest LGTM

**Pull request merged**